jetty-http2/http2-http-client-transport/src/test/java/org/eclipse/jetty/http2/client/http/InvocationTypeTest.java

```diff
@@ -0,0 +1,184 @@
+            {
+                serverEndPointRef.compareAndSet(null, (AbstractEndPoint)jettyRequest.getHttpChannel().getEndPoint());
+                // Write a large content to cause TCP congestion.
+                response.getOutputStream().write(new byte[contentLength]);
+            }
+        });
+
+        HTTP2Client client = new HTTP2Client();
+        // Set large flow control windows so the server hits TCP congestion.
+        int window = 2 * contentLength;
+        client.setInitialSessionRecvWindow(window);
+        client.setInitialStreamRecvWindow(window);
+        client.start();
+
+        FuturePromise<Session> promise = new FuturePromise<>();
+        client.connect(new InetSocketAddress("localhost", connector.getLocalPort()), new Session.Listener.Adapter(), promise);
+        Session session = promise.get(5, TimeUnit.SECONDS);
+
+        CountDownLatch clientBlockLatch = new CountDownLatch(1);
+        CountDownLatch clientDataLatch = new CountDownLatch(1);
+        // Send a request to TCP congest the server.
+        HttpURI uri1 = new HttpURI("http://localhost:" + connector.getLocalPort() + "/congest");
+        MetaData.Request request1 = new MetaData.Request("GET", uri1, HttpVersion.HTTP_2, new HttpFields());
+        session.newStream(new HeadersFrame(request1, null, true), new Promise.Adapter<>(), new Stream.Listener.Adapter()
+        {
+            @Override
+            public void onData(Stream stream, DataFrame frame, Callback callback)
+            {
+                // Block here to stop reading from the network
+                // to cause the server to TCP congest.
+                awaitUntil(0, () -> clientBlockLatch.await(5, TimeUnit.SECONDS));
+                callback.succeeded();
+                if (frame.isEndStream())
+                    clientDataLatch.countDown();
+            }
+        });
+
+        awaitUntil(5000, () ->
+        {
+            AbstractEndPoint serverEndPoint = serverEndPointRef.get();
+            return serverEndPoint != null && serverEndPoint.getWriteFlusher().isPending();
+        });
+        // Wait for NIO on the server to be OP_WRITE interested.
+        Thread.sleep(1000);
+
+        // Make sure there is a reserved thread.
+        serverThreads.tryExecute(() ->
+        {});
+        awaitUntil(5000, () -> serverThreads.getAvailableReservedThreads() == 1);
+        // Use the reserved thread for a blocking operation, simulating another blocking write.
+        CountDownLatch serverBlockLatch = new CountDownLatch(1);
+        assertTrue(serverThreads.tryExecute(() -> awaitUntil(0, () -> serverBlockLatch.await(15, TimeUnit.SECONDS))));
+
+        // Unblock the client to read from the network, which should unblock the server write().
+        clientBlockLatch.countDown();
+
+        assertTrue(clientDataLatch.await(10, TimeUnit.SECONDS), server.dump());
+        serverBlockLatch.countDown();
+    }
+
+    public void awaitUntil(long millis, Callable<Boolean> test)
```

Comment:
The await semantic is good for testing.... but let's not cut and paste it everywhere.  Can we either import the 3rd party util or add the method/class to our own tools.         OK not to fix in this PR, but can we create an issue that records to need to do it once and only once... so it will eventually be fixed.

@lorban what is the status of the await dependency?

Reply:
This is still duplicated